java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -92,11 +88,18 @@ private CallOption[] getOptions() {
    * @param flightInfo The {@link FlightInfo} instance from which to fetch results.
    * @return a {@code FlightStream} of results.
    */
-  public List<FlightStream> getStreams(final FlightInfo flightInfo) {
-    return flightInfo.getEndpoints().stream()
-        .map(FlightEndpoint::getTicket)
-        .map(ticket -> sqlClient.getStream(ticket, getOptions()))
-        .collect(Collectors.toList());
+  public List<FlightStream> getStreams(final FlightInfo flightInfo) throws RuntimeException {
+    try {
+      ArrayList<FlightStream> streams = new ArrayList<>();
+      for (FlightEndpoint ep : flightInfo.getEndpoints()) {
+        URI uri = ep.getLocations().size() > 0 ? ep.getLocations().get(0).getUri() : null;
+        FlightSqlClient sqlClient = this.connectionManager.getConnection(uri);
```

Comment:
Can we rename getConnection() to acquireConnection()? To make it more clear that it's a resource acquisition.